From abc8e0bb277f7a54ff4ee209845dde910d267819 Mon Sep 17 00:00:00 2001
From: William S Fulton <wsf@fultondesigns.co.uk>
Date: Fri, 27 May 2022 18:58:04 +0100
Subject: [PATCH] Define PY_SSIZE_T_CLEAN macro

before #include "Python.h" as recommended in Python 3.7 and later.

Issue #2277
---
 CHANGES.current          | 11 +++++++++++
 Lib/python/embed.i       |  2 +-
 Lib/python/pyruntime.swg |  2 +-
 3 files changed, 13 insertions(+), 2 deletions(-)

Index: swig-4.0.2/CHANGES.current
===================================================================
--- swig-4.0.2.orig/CHANGES.current
+++ swig-4.0.2/CHANGES.current
@@ -7,6 +7,17 @@ the issue number to the end of the URL:
 Version 4.0.2 (8 Jun 2020)
 ==========================
 
+2022-05-27: xypron
+            [Python] #2277 Define PY_SSIZE_T_CLEAN macro before #include "Python.h" as
+            recommended in Python 3.7 and later.
+
+            To avoid this macro definition, add the following to your interface file so
+            that SWIG_NO_PY_SSIZE_T_CLEAN is defined at the beginning of the C++ wrappers:
+
+              %begin %{
+              #define SWIG_NO_PY_SSIZE_T_CLEAN
+              %}
+
 2020-06-07  vigsterkr
             [Ruby] #1717 Nil fix mangling strings
 
Index: swig-4.0.2/Lib/python/embed.i
===================================================================
--- swig-4.0.2.orig/Lib/python/embed.i
+++ swig-4.0.2/Lib/python/embed.i
@@ -29,7 +29,7 @@ As far as I know, this module is C++ saf
 #endif
 
 %wrapper %{
-#ifndef PY_SSIZE_T_CLEAN
+#ifndef SWIG_NO_PY_SSIZE_T_CLEAN
 #define PY_SSIZE_T_CLEAN
 #endif
 #include <Python.h>
Index: swig-4.0.2/Lib/python/pyruntime.swg
===================================================================
--- swig-4.0.2.orig/Lib/python/pyruntime.swg
+++ swig-4.0.2/Lib/python/pyruntime.swg
@@ -4,7 +4,7 @@
 # include <math.h>
 #endif
 
-#ifndef PY_SSIZE_T_CLEAN
+#ifndef SWIG_NO_PY_SSIZE_T_CLEAN
 #define PY_SSIZE_T_CLEAN
 #endif
 #if defined(_DEBUG) && defined(SWIG_PYTHON_INTERPRETER_NO_DEBUG)
